src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/FeatureChainingWfsTest.java

```diff
@@ -1372,6 +1399,134 @@ public void testEncodeFeatureMembers() throws Exception {
+
+        JDBCDataStore store = (JDBCDataStore) rootMapping.getSource().getDataStore();
+        NestedFilterToSQL nestedFilterToSQL = createNestedFilterEncoder(rootMapping);
+
+        FilterFactoryImplNamespaceAware ff = new FilterFactoryImplNamespaceAware();
+        ff.setNamepaceContext(rootMapping.getNamespaces());
+
+        /*
+         * test combined filters on nested attributes
+         */
+        And and = ff
+                .and(ff.equals(ff.property("gsml:specification/gsml:GeologicUnit/gml:name"),
+                        ff.literal("New Group")),
+                        ff.equals(
+                                ff.property("gsml:specification/gsml:GeologicUnit/gsml:composition/gsml:CompositionPart/gsml:proportion/gsml:CGI_TermValue/gsml:value"),
+                                ff.literal("significant")));
+
+        // Each filter involves a single nested attribute --> can be encoded
+        ComplexFilterSplitter splitter = new ComplexFilterSplitter(store.getFilterCapabilities(),
+                rootMapping);
+        splitter.visit(and, null);
+        Filter preFilter = splitter.getFilterPre();
+        Filter postFilter = splitter.getFilterPost();
+
+        assertEquals(and, preFilter);
+        assertEquals(Filter.INCLUDE, postFilter);
+
+        // filter must be "unrolled" (i.e. reverse mapped) first
+        Filter unrolled = AppSchemaDataAccess.unrollFilter(and, rootMapping);
+
+        // Filter is nested
+        assertTrue(NestedFilterToSQL.isNestedFilter(unrolled));
+
+        String encodedFilter = nestedFilterToSQL.encodeToString(unrolled);
+
+        assertTrue(encodedFilter.matches("^\\(EXISTS.*AND EXISTS.*\\)$"));
+        assertContainsFeatures(fs.getFeatures(and), "mf4");
+
+        /*
+         * test like filter on nested attribute
+         */
+         PropertyIsLike like = ff.like(ff.property("gsml:specification/gsml:GeologicUnit/gml:description"), "*sedimentary*");
+
+        // Filter involving single nested attribute --> can be encoded
+        ComplexFilterSplitter splitterLike = new ComplexFilterSplitter(store.getFilterCapabilities(),
+                rootMapping);
+        splitterLike.visit(like, null);
+        preFilter = splitterLike.getFilterPre();
+        postFilter = splitterLike.getFilterPost();
+
+        assertEquals(like, preFilter);
+        assertEquals(Filter.INCLUDE, postFilter);
+
+        // filter must be "unrolled" (i.e. reverse mapped) first
+        unrolled = AppSchemaDataAccess.unrollFilter(like, rootMapping);
+
+        // Filter is nested
+        assertTrue(NestedFilterToSQL.isNestedFilter(unrolled));
+
+        encodedFilter = nestedFilterToSQL.encodeToString(unrolled);
+
+        // this is the generated query in PostGIS, but the test limits to check the presence of the
+        // EXISTS keyword, as the actual SQL is dependent on the underlying database
+//        String expected = "EXISTS (SELECT \"chain_link_1\".\"PKEY\" "
```

Comment:
I would prefer this commented code to be indented to match the other code. Yes, this is the Eclipse default, but this is not commented code: it is real in-code documentation!